mutiny-core/src/nostr/nwc.rs

```diff
@@ -27,11 +29,57 @@ pub struct SingleUseSpendingConditions {
     pub amount_sats: u64,
 }
 
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct TrackedPayment {
+    /// Time in seconds since epoch
+    time: u64,
+    /// Amount in sats
+    amt: u64,
+    /// Payment hash
+    hash: String,
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct BudgetedSpendingConditions {
+    /// Amount in sats for the allotted budget period
+    pub budget: u64,
+    /// Max amount in sats for a single payment
+    pub single_max: Option<u64>,
+    /// Payment history
+    pub payments: Vec<TrackedPayment>,
```

Comment:
We need some sort of lock around tracked payments and adding / removing them. I don't know if it's just a bug in our pending approval list or not but those are more miss than hit at the moment, when there's multiple involved.

Reply:
It is currently safe because we don't process NWCs concurrently but yeah, probably want that just to be safe

Reply:
This can be done in a follow up for later, but we'd also need to ensure some of the same guarantees for gifting too.